consensus/istanbul/backend/test_utils.go

```diff
@@ -85,14 +85,16 @@ func newBlockChainWithKeys(isProxy bool, proxiedValAddress common.Address, isPro
 	b.StartAnnouncing()
 
 	if !isProxy {
-		b.SetBlockProcessors(blockchain.HasBadBlock,
+		b.SetCallBacks(blockchain.HasBadBlock,
 			func(block *types.Block, state *state.StateDB) (types.Receipts, []*types.Log, uint64, error) {
 				return blockchain.Processor().Process(block, state, *blockchain.GetVMConfig())
 			},
 			func(block *types.Block, state *state.StateDB, receipts types.Receipts, usedGas uint64) error {
 				return blockchain.Validator().ValidateState(block, state, receipts, usedGas)
+			},
+			func(block *types.Block, receipts []*types.Receipt, logs []*types.Log, state *state.StateDB, emitHeadEvent bool) (status types.WriteStatus, err error) {
+				return blockchain.WriteBlockWithState(block, receipts, logs, state, emitHeadEvent)
 			})
```

Comment:
I think this could be reduced to:

```suggestion
			blockchain.Validator().ValidateState,
			blockchain.WriteBlockWithState)
```

Reply:
But actually if we change the callback method to not return the status as you suggested above we can't keep this simplification verbatim.